### DISCUSSION TITLE
How to hide `projectId` in production mode?

### DISCUSSION THREAD

**Discussion by dnnagy** [WalletConnect/walletconnect-monorepo#3680] [Developer Support] [answered]
I'm using Web3Modal in a react app, and I noticed, that the `projectId` is visible in the network tab at every API call:
![walletconnect](https://github.com/WalletConnect/walletconnect-monorepo/assets/12029086/770f8881-064b-49b2-aee8-07a51c1a8727)

Thus, an attacker could very easily extract my `projectId` and reuse it in a malicious way. What is the canonical method to hide `projectId` in a production app?

**Comment by glitch-txs** [comment, marked as answer]
Hi, the project ID is meant to be public, anything that's on the frontend side is public. However you can whitelist your domain in your cloud's project and you'll get CORS protection.

**Reply by jakubgs** [reply]
Then that should probably be indicated in the UI and/or the docs.
